internal/features/stacks/state/stack_store.go

```diff
@@ -253,6 +309,26 @@ func stackCopyByPath(txn *memdb.Txn, path string) (*StackRecord, error) {
 func (s *StackStore) queueRecordChange(oldRecord, newRecord *StackRecord) error {
 	changes := globalState.Changes{}
 
+	switch {
+	// new record added
+	case oldRecord == nil && newRecord != nil:
+		if newRecord.TerraformVersion != nil {
+			changes.TerraformVersion = true
+		}
+	// record removed
+	case oldRecord != nil && newRecord == nil:
+		changes.IsRemoval = true
+
+		if oldRecord.TerraformVersion != nil {
+			changes.TerraformVersion = true
+		}
+	// record changed
+	default:
+		if oldRecord.TerraformVersion == nil || !oldRecord.TerraformVersion.Equal(newRecord.TerraformVersion) {
```

Comment:
```suggestion
		if !oldRecord.TerraformVersion.Equal(newRecord.TerraformVersion) {
```

We don't need the `nil` check here, because `.Equal` already does it.

Reply:
Ah, TIL – I was unsure whether that could crash if `TerraformVersion` is nil 😅

Reply:
yeah, it sometimes feels weird to have methods available on `nil`